src/main.c

```diff
@@ -1,4 +1,26 @@
-int main () {
-	while(1);
+#include <stdio.h>
+#include "app/app_header.h"
+
+bool check_app_version(const app_header_t header) {
```

Comment:
I would move this functionality into a different file like `boot.c` and have this be a static function. You could create a function called `boot_app()` which takes a pointer to an `app_header_t` and then attempts to boot into the app. You can also define an enum with errors that it can return to signify to the caller that something went wrong. An example could be `BOOT_ERROR_INCOMPATIBLE_HEADER`.